spanner_dbapi/connection.py

```diff
@@ -5,131 +5,98 @@
-
-    def read_snapshot(self):
-        self.__raise_if_already_closed()
-        return self._dbhandle.snapshot()
-
-    def in_transaction(self, fn, *args, **kwargs):
-        self.__raise_if_already_closed()
-        return self._dbhandle.run_in_transaction(fn, *args, **kwargs)
-
-    def append_ddl_statement(self, ddl_statement):
-        self.__raise_if_already_closed()
-        self._ddl_statements.append(ddl_statement)
-
-    def run_prior_DDL_statements(self):
-        self.__raise_if_already_closed()
-
-        if not self._ddl_statements:
-            return
-
-        ddl_statements = self._ddl_statements
-        self._ddl_statements = []
-
-        return self.__handle_update_ddl(ddl_statements)
-
-    def list_tables(self):
-        return self.run_sql_in_snapshot(
-            """
-            SELECT
-              t.table_name
-            FROM
-              information_schema.tables AS t
-            WHERE
-              t.table_catalog = '' and t.table_schema = ''
-            """
-        )
 
-    def run_sql_in_snapshot(self, sql, params=None, param_types=None):
-        # Some SQL e.g. for INFORMATION_SCHEMA cannot be run in read-write transactions
-        # hence this method exists to circumvent that limit.
-        self.run_prior_DDL_statements()
-
-        with self._dbhandle.snapshot() as snapshot:
-            res = snapshot.execute_sql(
-                sql, params=params, param_types=param_types
-            )
-            return list(res)
-
-    def get_table_column_schema(self, table_name):
-        rows = self.run_sql_in_snapshot(
-            """SELECT
-                COLUMN_NAME, IS_NULLABLE, SPANNER_TYPE
-            FROM
-                INFORMATION_SCHEMA.COLUMNS
-            WHERE
-                TABLE_SCHEMA = ''
-            AND
-                TABLE_NAME = @table_name""",
-            params={"table_name": table_name},
-            param_types={"table_name": spanner.param_types.STRING},
-        )
+class Connection(object):
+    """This is a wrap-around object for the existing `Database` and the
+    corresponding `Instance` objects.
 
-        column_details = {}
-        for column_name, is_nullable, spanner_type in rows:
-            column_details[column_name] = ColumnDetails(
-                null_ok=is_nullable == "YES", spanner_type=spanner_type
-            )
-        return column_details
+    :type database: :class:`~google.cloud.spanner_v1.database.Database`
+    :param database: Corresponding Database.
 
-    def close(self):
-        self.rollback()
-        self.__dbhandle = None
-        self._closed = True
+    :type instance: :class:`~google.cloud.spanner_v1.instance.Instance`
+    :param instance: The instance that owns the database.
+    """
 
-    def commit(self):
-        self.__raise_if_already_closed()
+    def __init__(self, database, instance):
+        self.database = database
+        self.instance = instance
```

Comment:
Same here, can just use
```
    def __init__(self, database):
        self.database = database
```
and then referring to `instance` via `self.database._instance`.